runtime/src/lib.rs

```diff
@@ -101,6 +103,61 @@ impl Runtime {
     }
 }
 
+struct EnvAdapter<'a>{
+    system: &'a mut dyn System,
+}
+
+impl Environment for EnvAdapter<'_> {
+
+    fn elapsed(&self) -> Result<Duration, Error> {
+        unimplemented!();
```

Comment:
I'm guessing we'll need to store away some sort of epoch that all calls to `elapsed()` are done relative to. Could we give the `WasmProgram` a `created_at: Instant` field when it gets constructed, then give `EnvAdapter` a copy so `elapsed()` just defers to `self.created_at.elapsed()`?

Reply:
Yes, that sounds reasonable. There's all sorts of stuff needing a clock, like time delay blocks and so on.

Reply:
Attempted in 8baa4dd